File: 20.Valid_Parentheses/main.py
class Solution:
    def is_valid(self, s: str) -> bool:
        stack = []
        for char in s:
            if char in "({[":
                stack.append(char)  # Push to stack
            else:
                if not stack:
                    return False
                top = stack.pop()  # Pop from stack
                if (char == ')' and top != '(') or \
                   (char == '}' and top != '{') or \
                   (char == ']' and top != '['):
                    return False
        return not stack

if __name__ == "__main__":
    solution = Solution()
    s = "({[]})"
    print("Is Valid:", solution.is_valid(s))  # Output: True

File: 20.Valid_Parentheses/main.ts
class Solution {
     isValid(s: string): boolean {
         const stack: string[] = [];
         for (const char of s) {
             if (char === '(' || char === '{' || char === '[') {
                 stack.push(char); // Push to stack
             } else {
                 if (stack.length === 0) return false;
                 const top = stack.pop()!;
                 if ((char === ')' && top !== '(') || 
                     (char === '}' && top !== '{') || 
                     (char === ']' && top !== '[')) {
                     return false;
                 }
             }
         }
         return stack.length === 0;
     }
 }
 
 const solution = new Solution();
 const s = "({[]})";
 console.log("Is Valid:", solution.isValid(s)); // Output: True
 